### PR TITLE
refactor: integrated base signer structure into existing EVM/BTC signers

### PR DESCRIPTION
**Pull request by ws4charlie**
# Description

The PR integrated base `Signer` structures into existing EVM/BTC `Signer` structures.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [x] Tested CCTX in localnet
- [ ] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -48,6 +48,7 @@
 * [2296](https://github.com/zeta-chain/node/pull/2296) - move `testdata` package to `testutil` to organize test-related utilities
 * [2344](https://github.com/zeta-chain/node/pull/2344) - group common data of EVM/Bitcoin signer and observer using base structs
 * [2317](https://github.com/zeta-chain/node/pull/2317) - add ValidateOutbound method for cctx orchestrator
+* [2357](https://github.com/zeta-chain/node/pull/2357) - integrated base Signer structure into EVM/Bitcoin Signer
```
Comment (lumtis) [inline changelog.md:51]:
```suggestion
* [2357](https://github.com/zeta-chain/node/pull/2357) - integrate base Signer structure into EVM/Bitcoin Signer
```

File: zetaclient/chains/evm/signer/signer.go [hunk 2]
```diff
@@ -48,39 +47,57 @@ var (
...
+	zetaConnectorAddress ethcommon.Address
+
+	// er20CustodyAddress is the address of the ERC20Custody contract
+	er20CustodyAddress ethcommon.Address
+
+	// outboundHashBeingReported is a map of outboundHash being reported
 	outboundHashBeingReported map[string]bool
+
+	// mu protects fields from concurrent access
+	mu *sync.Mutex
```
Comment (lumtis) [inline zetaclient/chains/evm/signer/signer.go:75]: Couldn't it be part of base signer as well?

**Review by lumtis — APPROVED** [review]
👍

File: zetaclient/chains/evm/signer/signer.go [hunk 3]
```diff
@@ -529,22 +538,21 @@ func (signer *Signer) BroadcastOutbound(
...
 	}
-	// Try to broadcast transaction
+
+	// broadcast transaction
 	if tx != nil {
 		outboundHash := tx.Hash().Hex()
-		logger.Info().
-			Msgf("on chain %s nonce %d, outboundHash %s signer %s", signer.chain, cctx.GetCurrentOutboundParam().TssNonce, outboundHash, myID)
-		//if len(signers) == 0 || myid == signers[send.OutboundParams.Broadcaster] || myid == signers[int(send.OutboundParams.Broadcaster+1)%len(signers)] {
+		logger.Info().Msgf("BroadcastOutbound: broadcasting tx %s on chain %d nonce %d signer %s",
```
Comment (kingpinXD) [inline zetaclient/chains/evm/signer/signer.go:545]: I am wondering if we should move this to debug. 
The broadcasted log is already in the info. Do we need both ?

File: zetaclient/testutils/os.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+// create a temporary directory for testing
+func CreateTempDir(t *testing.T) string {
+	tempPath, err := os.MkdirTemp("", "tempdir-")
+	require.NoError(t, err)
+	return tempPath
+}
```
Comment (kingpinXD) [inline zetaclient/testutils/os.go:15]: NIT: Why not put this is the sample directory?

**Review by kingpinXD — APPROVED** [review]
lgtm

File: zetaclient/chains/evm/signer/signer.go (hunk 2 above)
Comment (ws4charlie) [reply zetaclient/chains/evm/signer/signer.go:75]: Having a mutex in base-level struct is completely possible. Keep in mind, only the sub-class (now the base class) knows when it needs to be used. Let's do it and add a description.

Comment (ws4charlie) [reply zetaclient/chains/evm/signer/signer.go]: https://github.com/zeta-chain/node/pull/2357/commits/e585bc0249a23d140a11e94c4be327d0075a6e8c

File: zetaclient/testutils/os.go (hunk 4 above)
Comment (ws4charlie) [reply zetaclient/testutils/os.go]: https://github.com/zeta-chain/node/pull/2357/commits/e585bc0249a23d140a11e94c4be327d0075a6e8c

**Pull request merged**